opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -263,7 +263,94 @@ type LogsData = Box<(LogRecord, InstrumentationScope)>;
 /// let provider = LoggerProvider::builder()
 ///     .with_log_processor(processor)
 ///     .build();
+/// ```
+///
+/// **Memory Management in BatchLogProcessor**
+///
+/// The `BatchLogProcessor` manages memory through the following stages of log processing:
+///
+/// 1. **Record Ingestion**:
+///    - Each `LogRecord` is **cloned** upon entering the processor.
+///    - `LogRecordAttributes` utilize a hybrid memory model:
```

Comment:
How attributes are stored inside LogRecord is not a concern of BatchLogProcessor. It can be documented (and should be!), but not in processor doc.